File: content-solutions/two-pointers/sorted-squares.py
def solution(nums):
    n = len(nums)
    result = [0] * n
    l, r, k = 0, n - 1, n - 1
    while l <= r:
        if abs(nums[l]) > abs(nums[r]):
            result[k] = nums[l] ** 2
            l += 1
        else:
            result[k] = nums[r] ** 2
            r -= 1
        k -= 1
    return result

File: content-solutions/two-pointers/sorted-squares.js
function solution(nums) {
  const n = nums.length;
  const result = Array(n).fill(0);
  let l = 0;
  let r = n - 1;
  for (let k = n - 1; k >= 0; k--) {
    if (Math.abs(nums[l]) > Math.abs(nums[r])) {
      result[k] = nums[l] * nums[l];
      l++;
    } else {
      result[k] = nums[r] * nums[r];
      r--;
    }
  }
  return result;
}
